Sources/Phoenix/Socket.swift

```diff
@@ -489,7 +509,13 @@ extension Socket {
                 assertionFailure("We are not currently expecting any data frames from the server")
             case .text(let string):
                 do {
-                    let message = try IncomingMessage(string: string)
+                    let message: IncomingMessage
+                    if let decoder = self.customDecoder {
+                        message = try decoder(try string.data(using: .utf8))
+                    } else {
+                        message = try IncomingMessage(string: string)
```

Comment:
@atdrendel why not set IncomingMessage as the default customDecoder and then just always defer to what's set?

Reply:
That's a good idea. Much cleaner.